File: app/src/main/java/agersant/polaris/features/browse/BrowseAdapterDiscography.java
package agersant.polaris.features.browse;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import agersant.polaris.PolarisService;
import agersant.polaris.R;


class BrowseAdapterDiscography extends BrowseAdapter {

	BrowseAdapterDiscography(PolarisService service) {
		super(service);
	}

	@Override
	public BrowseItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		View itemQueueStatusView = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_browse_item_queued, parent, false);
		View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_browse_discography_item, parent, false);
		return new BrowseItemHolderDiscography(service, this, itemView, itemQueueStatusView);
	}

}

File: app/src/main/java/agersant/polaris/features/browse/BrowseAdapterDiscography.kt
package agersant.polaris.features.browse;

import agersant.polaris.PlaybackQueue
import agersant.polaris.api.API
import agersant.polaris.databinding.ViewBrowseDiscographyItemBinding
import agersant.polaris.databinding.ViewQueueStatusBinding
import android.view.LayoutInflater
import android.view.ViewGroup


internal class BrowseAdapterDiscography(
    private val api: API,
    private val playbackQueue: PlaybackQueue,
) : BrowseAdapter() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BrowseItemHolder {
        val inflater = LayoutInflater.from(parent.context)
        val itemBinding = ViewBrowseDiscographyItemBinding.inflate(inflater, parent, false);
        val itemQueueStatusView = ViewQueueStatusBinding.inflate(inflater, parent, false);

        return BrowseItemHolderDiscography(api, playbackQueue, this, itemBinding, itemQueueStatusView);
    }
}
